astropy/coordinates/matrix_utilities.py

```diff
@@ -61,10 +61,13 @@ def rotation_matrix(angle, axis='z', unit=None):
     rmat : `numpy.matrix`
         A unitary rotation matrix.
     """
-    if unit is None:
-        unit = u.degree
-
-    angle = Angle(angle, unit=unit)
+    if isinstance(angle, u.Quantity):
+        angle = angle.to_value(u.radian)
+    else:
+        if unit is None:
+            angle = np.deg2rad(angle)
+        else:
+            angle = u.Unit(unit).to(u.rad, angle)
```

Comment:
If passed in a float, this one will return a float

Reply:
You are right.